### PR TITLE
Add docker and internal cron scheduler

### PR DESCRIPTION
**Pull request by andytson**
I prefer running these sorts of things in docker, with their own scheduler

### CODE REVIEW

**Comment by paulmaunders** [comment]
Could you re-base these changes against the current version as there are a few conflicts at the mo?

File: delivery-slots.js [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
+const shell = require("shelljs");
 
 // Read config
 const config = ini.parse(fs.readFileSync("./config.ini", "utf-8"));
 
-// Log time
-var executiontime = Date.now();
-var date = new Date(executiontime);
-console.log(executiontime + ' ' + date.toUTCString());
+function getBrowser() {
```
Comment (paulmaunders) [inline delivery-slots.js]: What's the advantage of doing compared to the existing way?

File: delivery-slots.js [hunk 2]
```diff
@@ -110,4 +113,13 @@ async function run() {
   }
 }
 
-run();
+if (config.cron) {
+  schedule.scheduleJob(config.cron, () => run());
```
Comment (paulmaunders) [inline delivery-slots.js]: Can we still keep a log when it runs on the cron?

File: delivery-slots.js (hunk 1 above)
Comment (andytson) [reply delivery-slots.js]: Its awkward installing all the x11 dependencies for chromium to work in the node docker container. This allows instead a specifically suited docker image that has everything needed preinstalled. It does mean that memory for the browser is used for longer, but shouldn't be much of a problem with no pages open

File: delivery-slots.js (hunk 2 above)
Comment (andytson) [reply delivery-slots.js]: this is an outdated version, the new version logs the execution time for both modes.

Comment (paulmaunders) [reply delivery-slots.js]: Do you need to do a new pull request to get the newer version commit in?

Comment (andytson) [reply delivery-slots.js]: no, you should see the latest version in the files tab, specifically this is where the run time log is at now https://github.com/paulmaunders/delivery-slot-bot/pull/2/files#diff-82828ea2d34b83ab67542f306b66de78R24-R27

**Comment by andytson** [comment]
not sure if this is a bug in the PR or was already present, but I got alerts for each week even though only one week had slots

**Comment by andytson** [comment]
ah, spotted it. It's an existing bug that happens only if the first week has slots (as the `html` variable is only looking at the first week

**Pull request merged**